src/components/Vote.vue

```diff
@@ -1,26 +1,60 @@
 <template>
-  <div class="hello">
+  <div class="poll" v-loading="isLoading">
+    <el-card class="box-card" v-if="!this.isLoading">
+      <div v-if="!this.isVoted">
+        <el-button type="text" @click="vote(poll, 'first')">{{poll.first.value}}</el-button>
+        <el-button type="text" @click="vote(poll, 'second')">{{poll.second.value}}</el-button>
+      </div>
+      <el-row v-else>
+        <results :poll="poll"></results>
+      </el-row>
+    </el-card>
   </div>
 </template>
 <script>
 import db from '../services/firebase';
+import Results from './Results.vue';
 
 const polls = db.ref('/');
 
 export default {
   name: 'Vote',
   data() {
     return {
-      poll: {},
+      isLoading: true,
+      isVoted: false,
     };
   },
+  components: {
+    Results,
+  },
   firebase: {
-    polls,
+    polls: {
+      source: polls,
+      readyCallback() {
+        this.isLoading = false;
+        this.poll = this.polls.filter(poll => poll['.key'] === this.$route.params.id)[0];
+      },
+    },
   },
   methods: {
-    vote: () => {
+    vote(poll, choice) {
+      this.isVoted = true;
+
+      this.poll[choice].votes += 1;
 
+      db.ref(`/${this.$route.params.id}/${choice}`).set({
```

Comment:
What if two people vote at the same time very quickly? Isn't it possivle that some votes will be overridden?

More reliable way is to use `db.ref.transaction` like in this example: https://github.com/firebase/functions-samples/blob/3f4b76bf9d9893060b2937d47a645f1c396ec169/child-count/functions/index.js#L23-L39

Reply:
🤔 Valid point. I'll fix it asap.